app/gilded-rose.ts

```diff
@@ -19,51 +19,70 @@ export class GildedRose {
 
   updateQuality() {
     for (let i = 0; i < this.items.length; i++) {
-      if (this.items[i].name != 'Aged Brie' && this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-        if (this.items[i].quality > 0) {
-          if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-            this.items[i].quality = this.items[i].quality - 1
-          }
-        }
-      } else {
-        if (this.items[i].quality < 50) {
-          this.items[i].quality = this.items[i].quality + 1
-          if (this.items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].sellIn < 11) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
-            }
-            if (this.items[i].sellIn < 6) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
-            }
-          }
-        }
+      if(!this.items[i].name.includes("Sulfuras",0)){
+        this.items[i].sellIn -= 1;
+        this.applyQualityChange(this.items[i], this.getItemQualityChange(this.items[i]));
       }
-      if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-        this.items[i].sellIn = this.items[i].sellIn - 1;
-      }
-      if (this.items[i].sellIn < 0) {
-        if (this.items[i].name != 'Aged Brie') {
-          if (this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].quality > 0) {
-              if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-                this.items[i].quality = this.items[i].quality - 1
-              }
-            }
-          } else {
-            this.items[i].quality = this.items[i].quality - this.items[i].quality
-          }
-        } else {
-          if (this.items[i].quality < 50) {
-            this.items[i].quality = this.items[i].quality + 1
-          }
-        }
+    }
+    return this.items;
+  }
+
+  getItemQualityChange(item:Item){
+    let change = this.getQualityChange(item);
+    if(item.name.includes("Backstage passes")) {
+      change = this.getPassesQualityChange(item);
+    }
+    else if(item.name === "Aged Brie"){
+      change = -change;
+    }
```

Comment:
Do you agree that it's safe to assume an item can't be both "backstage passes" and "aged brie"? If so, you can separate out the else if block into an if block and that might make your "conjured" if block nicer since you can move the "aged brie" if block to after the "conjured" if block, flipping the sign only once

Reply:
I was considering a case of something like "Backstage passes made of Aged Brie" or something along those lines, but the interaction between those seems somewhat arbitrary, and I didn't want to extend the scope too much